### PR TITLE
Bug 1021 tab logic v2

### PR DESCRIPTION
**Pull request by ethanstrominger**
I would like to do more refactoring of this, including:

- see if any duplicated code in OfferPostsContainer and ReqestPostsContainer can be combined without sacrificing readability of either
- add tests

If you don't see anything significantly wrong and you think this improves the code since it fixes a bug, can you merge? I can make changes you want.

### CODE REVIEW

**Review by ashwinkjoseph — COMMENTED** [review]
I notice the need for changing the names at a few places, I will add a more detailed review of this tomorrow

File: web-client/src/ducks/requests/functions.ts
Comment (ashwinkjoseph) [inline web-client/src/ducks/requests/functions.ts:55]: can you make use of firebase directly here to get the data rather than the cloud function? that is what we are aiming for, so we shouldn't use the cloud function here.

File: web-client/src/ducks/requests/actions.ts
Comment (ashwinkjoseph) [inline web-client/src/ducks/requests/actions.ts:27]: This will only give you the posts that have status as open. We need posts of all statuses here. I have already mentioned as separate comment in the functions file for using firebase directly.

File: web-client/src/ducks/requests/actions.ts
Comment (ashwinkjoseph) [inline web-client/src/ducks/requests/actions.ts:36]: This would only give you posts which you created which have been accepted. This function is supposed to give you the posts for which you responded as well.
I have already mentioned as a separate comment on the functions page for using firebase directly instead of the cloud functions

**Review by ashwinkjoseph — CHANGES_REQUESTED** [review]
Can you switch to using firebase directly here?
There are a few changes to variable names that are required as well, I will put details regarding those tomorrow but they aren't super important, I can make them as a separate PR after merging this if you would like to proceed in that manner or we can wait till tomorrow and make the changes in this PR itself.

However, the functions should make use of firebase directly and should no longer make use of the cloud functions

File: web-client/src/ducks/requests/functions.ts
Comment (ethanstrominger) [reply web-client/src/ducks/requests/functions.ts:55]: Yes, I am working on that, but would like it to be a separate pull request.

File: web-client/src/ducks/requests/actions.ts
Comment (ethanstrominger) [reply web-client/src/ducks/requests/actions.ts:27]: I know, I want to do this as part of a separate pull request.  What is in develop branch has same issue.

File: web-client/src/ducks/requests/actions.ts
Comment (ethanstrominger) [reply web-client/src/ducks/requests/actions.ts:36]: I know, I want to do this as part of a separate pull request. What is in develop branch has same issue.

**Review by ashwinkjoseph — APPROVED** [review]
Approving with the knowledge of changing in another PR

**Pull request merged**
